### PR TITLE
32 error handling glue python

### LINKED ISSUE #32: Error Handling Glue -- Python
Implement the strategy in #22:

- [ ] Implement a set of Python Exception classes to mirror the Rust error cases.
- [ ] Write glue to check for errors returned through FFI, raise Python Exceptions.

### PR DESCRIPTION
**Pull request by andrewvyrros**
Closes #32.

### CODE REVIEW

**Comment by andrewvyrros** [comment]
Not the cleanest solution ever, but this gives us exceptions and rudimentary backtraces in Python.

File: rust/opendp-ffi/src/core.rs [hunk 1]
```diff
@@ -38,6 +35,79 @@ impl FfiObject {
...
+            || (error_string.clone(), None),
+            |i| (substring(&error_string, 0, i), Some(substring(&error_string, i + 2, error_string.len() - 2)))
+        );
+        let backtrace = format!("{:?}", error.backtrace);
+        let ffi_error = FfiError {
+            variant: util::into_c_char_p(type_message.0),
+            message: type_message.1.map_or(ptr::null::<c_char>() as *mut c_char, util::into_c_char_p),
+            backtrace: util::into_c_char_p(backtrace)
+        };
+        util::into_raw(ffi_error)
```
Comment (Shoeboxam) [inline rust/opendp-ffi/src/core.rs:63]: I did some reworking of the error type in `32-error-handling-glue-python-macros` to clean this up. Up to you if it goes in this PR.

**Review by Shoeboxam — APPROVED** [review]

File: rust/opendp-ffi/src/core.rs (hunk 1 above)
Comment (Shoeboxam) [reply rust/opendp-ffi/src/core.rs:63]: Specifically, here: https://github.com/opendp/opendp/commit/651219dee278be6394f26936319b7bf438ef46d0#diff-126584c448e8cd1780d21068c8134f1e3dd2f4a89e73d93f466f11d4e9a96f41L46-R52

Comment (andrewvyrros) [reply rust/opendp-ffi/src/core.rs]: I think this is a nice improvement, I've merged the changes.

**Review by Shoeboxam — APPROVED** [review]

**Pull request merged**